### PR TITLE
Feat/templating add self-running templates

### PR DESCRIPTION
**Pull request by dwertheimer**
Imagine an xcallback that gives users the ability to "run" any template to perform an action...
Here's a simple example which will open today's note, add some text under a certain heading. All passed through callback and hitting a user-created Template. No plugin code required.

1. Create a template like this:
```
---
title: cdjournal
openNoteTitle:  <TODAY>
writeUnderHeading: "📝 Notes & Goals"
location: prepend
---
**<%- date.timestamp() %>**
<%- journalEntry %>
```
2. Run this link
```
noteplan://x-callback-url/runPlugin?pluginID=np.Templating&command=templateRunner&arg0=cdjournal&arg1=true&arg2=journalEntry%3dYes%20IT%20WORKED
```
3. It should open today's daily note and you should see text under your Notes & Goals heading

Known issue: If the heading doesn't exist, it's not working. I think this is an API bug...still chasing it.

### CODE REVIEW

File: np.Templating/plugin.json
Comment (mikeerickson) [inline np.Templating/plugin.json:111]: This goes against the naming conventions used for all commands.  You have it right in the previous command.

e.g.

"name: "np:tr"

File: np.Templating/plugin.json
Comment (mikeerickson) [inline np.Templating/plugin.json:113]: you could use "templateRunner" here an alias

File: np.Templating/src/Templating.js
Comment (mikeerickson) [inline np.Templating/src/Templating.js:346]: Rephase to use `unique` instead of `distinct` more "human", less "techy" :-)

**Review by mikeerickson — CHANGES_REQUESTED** [review]
All in all, seems pretty clean and straight forward.  I will work on getting this implemented after the 1.2 release though so that I can have proper testing time (I will be releasing 1.2.0-beta.01 today, had to wait until latest NotePlan Alpha to make sure the `invokePluginCommandByName` bug was fixed (which it was)

File: np.Templating/plugin.json
Comment (dwertheimer) [reply np.Templating/plugin.json:111]: Thx for the review @mikeerickson. Great comments. w/r/t the commands, I did this on purpose though. When we are finished testing, I plan to hide these commands from the UX, so no user will ever see them as `np:*` commands. But I needed a reasonable command name for users to use in the x-callback: `noteplan://x-callback-url/runPlugin?pluginID=np.Templating&command=templateRunner`, and I'm a bit afraid of colons in the xcallback context. Does that make sense or do you still want me to change it? `noteplan://x-callback-url/runPlugin?pluginID=np.Templating&command=np:tr`

File: np.Templating/src/Templating.js
Comment (dwertheimer) [reply np.Templating/src/Templating.js:346]: agreed. done.

File: np.Templating/src/Templating.js
Comment (dwertheimer) [reply np.Templating/src/Templating.js:356]: done

File: np.Templating/src/Templating.js
Comment (dwertheimer) [reply np.Templating/src/Templating.js:285]: Yes, that's fair. Was moving too fast in refactoring. Will do. BTW, this is 5 params. Would you prefer that I change the signature to collapse some of these into an options object?

File: np.Templating/CHANGELOG.md
Comment (dwertheimer) [reply np.Templating/CHANGELOG.md:97]: OK I will wait until you push again and merge. Pretty sure I pulled main before I started work on this.

File: np.Templating/src/Templating.js
Comment (dwertheimer) [reply np.Templating/src/Templating.js:317]: done

**Comment by dwertheimer** [comment]
Made those changes and pushed an updated version. No hurry @mikeerickson. Get to it whenever.

File: np.Templating/CHANGELOG.md
Comment (mikeerickson) [reply np.Templating/CHANGELOG.md:113]: I am curious where the "old" `CHANGELOG.md` content is coming from, it is surely current on the current repo?

https://github.com/NotePlan/plugins/blob/main/np.Templating/CHANGELOG.md

File: np.Templating/plugin.json
Comment (mikeerickson) [reply np.Templating/plugin.json:118]: Ah gotcha, then nothing to see here :-)

**Comment by dwertheimer** [comment]
@mikeerickson Bummer to have these PRs open so long cuz I am super bad at merging and now I don't know how I would merge this.

**Comment by mikeerickson** [comment]
Implemented in next version, cherry-picking code as opposed to merging PR as too many merge conflicts with current np.Templating code, which is diverged from the current release in which this PR was branched.

**Pull request closed without merging**
